container/libcontainer/helpers.go

```diff
@@ -178,9 +178,11 @@ func GetStats(cgroupManager cgroups.Manager, rootFs string, pid int, ignoreMetri
 
 	return stats, nil
 }
-func schedulerStatsFromProcs(rootFs string, pids []int) (info.CpuSchedstat, error) {
+func schedulerStatsFromProcs(rootFs string, pids []int, pidMetricsCache map[int]info.CpuSchedstat) (info.CpuSchedstat, error) {
+	seen := make(map[int]bool, len(pids))
```

Comment:
You dont need the `seen` variable.  Just aggregate everything in the podMetricsCache at the end.